lib/WebpackErrorArray.js

```diff
@@ -0,0 +1,33 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Andrew Powell <andrew@shellscape.org>
+*/
+"use strict";
+
+const WebpackError = require("./WebpackError");
+
+module.exports = class WebpackErrorArray extends Array {
+	constructor(...args) {
+		// TODO: validation will be enabled in a later PR
+		// if (args.length > 0 && typeof args[0] !== "number") {
+		// 	WebpackErrorArray.validate(...args);
+		// }
+		super(...args);
+	}
+
+	push(...args) {
+		// TODO: validation will be enabled in a later PR
+		// WebpackErrorArray.validate(...args);
+		super.push(...args);
+	}
+
+	static validate(...args) {
+		for (const arg of args) {
+			if (!(arg instanceof WebpackError)) {
```

Comment:
This will be a long transition to plugin authors. We must be prepared to receive a lot of issues about this message.

Reply:
Yep. Growing is hard. But if you've seen the garbage that plugin authors are pushing to the errors and warnings arrays... It'll be worth it.

Reply:
I have no idea about that! Could you share your _discoveries_? It may be a good idea to brew around what they are trying to do and how to provide a better DX.

Reply:
I sure can. Most of that was discovered while working on `webpack-stylish`. I'll add examples of those to the RFC thread this week for reference. There's already a lot of code in `webpack-stylish` that attempts to deal with the different kinds of data (strings, arrays, Errors) that get added to those collections. One more reason locking this down will really help the project and ecosystem in the future.